Recurly/BaseClient.cs

```diff
@@ -45,11 +49,38 @@ public class BaseClient {
             RestClient.AddDefaultHeader("Content-Type", "application/json");
         }
 
-        public IRestResponse<T> MakeRequest<T>(Method method, string url, Request body = null, Dictionary<string, object> queryParams = null) where T: new() {
+        public async Task<T> MakeRequestAsync<T>(Method method, string url, Request body = null, Dictionary<string, object> queryParams = null, CancellationToken cancellationToken = default(CancellationToken)) where T: new() {
             Debug.WriteLine($"Calling {url}");
+            var request = BuildRequest(method, url, body, queryParams);
+            var task = RestClient.ExecuteTaskAsync<T>(request, cancellationToken);
+            return await task.ContinueWith(t => {
+                var resp = t.Result;
+                this.HandleResponse(resp);
+                return resp.Data;
+            });
+        }
+
+        public T MakeRequest<T>(Method method, string url, Request body = null, Dictionary<string, object> queryParams = null) where T: new() {
+            Debug.WriteLine($"Calling {url}");
+            var request = BuildRequest(method, url, body, queryParams);
+            var resp = RestClient.Execute<T>(request);
+            this.HandleResponse(resp);
+            return resp.Data;
+        }
+
+        public void _SetApiUrl(string uri) {
+          if (System.Environment.GetEnvironmentVariable("RECURLY_INSECURE") == "true") {
+            Console.WriteLine("[SECURITY WARNING] _SetApiUrl is for testing only and not supported in production.");
+            this.RestClient.BaseUrl = new Uri(uri);
+          } else {
+            Console.WriteLine("[SECURITY WARNING] _SetApiUrl is for testing only and not supported in production.");
```

Comment:
We should pull out the duplicate `Console.WriteLine()` call to outside the `if/else` block since we're printing it either way

Reply:
went ahead and changed.